src/ucp/tag/offload/rndv.c

```diff
@@ -0,0 +1,303 @@
+/**
+ * Copyright (C) 2022, NVIDIA CORPORATION & AFFILIATES. ALL RIGHTS RESERVED.
+ * See file LICENSE for terms.
+ */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include <ucp/proto/proto_single.inl>
+#include <ucp/rndv/proto_rndv.inl>
+#include <ucp/tag/offload.h>
+#include <ucp/tag/tag_rndv.h>
+
+
+static UCS_F_ALWAYS_INLINE int
+ucp_tag_rndv_offload_supported(const ucp_proto_init_params_t *init_params,
+                               ucp_proto_rndv_ctrl_init_params_t *params)
+{
+    ucp_context_h context = init_params->worker->context;
+    ucp_lane_index_t lane = init_params->ep_config_key->tag_lane;
+    ucp_md_index_t md_index;
+    const uct_md_attr_t *md_attr;
+
+    /* Multirail is not supported with tag offload rndv, need to register
+     * memory on tag lane only. */
+    md_index = ucp_proto_common_get_md_index(init_params, lane);
+    md_attr  = &context->tl_mds[md_index].attr;
+
+    /* Not supported if can't register the buffer on tag lane MD (e.g. can't
+     * send GPU memory if no GPU_DIRECT_RDMA configured). */
+    if (!(md_attr->cap.reg_mem_types & UCS_BIT(params->mem_info.type)) ||
```

Comment:
this logic should be done by ucp_proto_common_find_lanes, when we derive the rndv/offload protocol from ucp_proto_single_init